tests/scripts/all.sh

```diff
@@ -2183,33 +2183,141 @@ component_test_psa_crypto_config_reference_ecdsa_use_psa () {
 component_test_psa_crypto_config_accel_ecdh () {
     msg "test: MBEDTLS_PSA_CRYPTO_CONFIG with accelerated ECDH"
 
+    # Algorithms and key types to accelerate
+    loc_accel_list="ALG_ECDH KEY_TYPE_ECC_KEY_PAIR KEY_TYPE_ECC_PUBLIC_KEY"
+
+    # Configure and build the test driver library
+    # -------------------------------------------
+
     # Disable ALG_STREAM_CIPHER and ALG_ECB_NO_PADDING to avoid having
     # partial support for cipher operations in the driver test library.
     scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_STREAM_CIPHER
     scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_ECB_NO_PADDING
 
-    loc_accel_list="ALG_ECDH KEY_TYPE_ECC_KEY_PAIR KEY_TYPE_ECC_PUBLIC_KEY"
     loc_accel_flags=$( echo "$loc_accel_list" | sed 's/[^ ]* */-DLIBTESTDRIVER1_MBEDTLS_PSA_ACCEL_&/g' )
     make -C tests libtestdriver1.a CFLAGS=" $ASAN_CFLAGS $loc_accel_flags" LDFLAGS="$ASAN_CFLAGS"
 
+    # Configure and build the main libraries
+    # --------------------------------------
+
+    # Start from default config (no USE_PSA or TLS 1.3) + driver support
     scripts/config.py set MBEDTLS_PSA_CRYPTO_DRIVERS
     scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
-    scripts/config.py unset MBEDTLS_USE_PSA_CRYPTO
-    scripts/config.py unset MBEDTLS_SSL_PROTO_TLS1_3
+
+    # Disable the module that's accelerated
     scripts/config.py unset MBEDTLS_ECDH_C
+
+    # Disable things that depend on it
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
```

Comment:
This partially overlaps `config_psa_crypto_config_ecdh_use_psa`. I would suggest having a common setup function with two parameters: driver_only, and use_psa. Don't we want to test with full config `-use_psa` here, to be as close as possible to the reference / reference with acceleration?

Reply:
I think the overlap is only temporary and will go away as we expand support for driver-only ECDH. Also, in the future we might need to remove `ECP_RESTARTABLE` in the driver-reference pair while we could keep it enabled here. So, I'd rather keep the config function shared only between components that are going to be compared with `analyzes_outcomes.py`.

Regarding the point about doing full minus `USE_PSA` here, yes, I agree that would be better, but that's pre-existing, and not specific to the ECDH component. At some point in the future we should rationalize our choices of configuration. I'm not sure testing with default ever makes sense (except when specifically testing the things work in the default config). Are you OK with leaving this for later?

Reply:
Sure!